g2p/mappings/__init__.py

```diff
@@ -176,34 +178,27 @@ def process_model_specs(self):  # noqa: C901
             self.rules = sorted(
                 # Temporarily normalize to NFD for heuristic sorting of NFC-defined rules
                 self.rules,
-                key=lambda x: (
-                    len(normalize(strip_index_notation(x.rule_input), "NFD"))
-                    if isinstance(x, Rule)
-                    else len(normalize(x["in"], "NFD"))
-                ),
+                key=lambda x: len(normalize(strip_index_notation(x.rule_input), "NFD")),
                 reverse=True,
             )
 
+        def apply_to_attributes(rule: Rule, func: Callable, *attrs):
+            for k in attrs:
+                value = getattr(rule, k)
+                if value:  # won't be None since default is ""
+                    setattr(rule, k, func(value))
+
         non_empty_mappings: List[Rule] = []
         for i, rule in enumerate(self.rules):
-            if isinstance(rule, dict):
-                rule = Rule(**rule)
             # Expand Abbreviations
-            if (
-                self.abbreviations
-                and self.rules
```

Comment:
**obviously* self.rules has non-zero length here because we are iterating over it